Lib/fontTools/varLib/__init__.py

```diff
@@ -130,16 +132,26 @@ def _add_fvar(font, axes, instances: List[InstanceDescriptor]):
         psname = instance.postScriptFontName
 
         inst = NamedInstance()
+        inst.coordinates = {
+            axes[k].tag: axes[k].map_backward(v) for k, v in coordinates.items()
+        }
+
+        # Instances should only reuse name ID 2 or 17 if they are at the default
+        # location across all axes. See
+        # https://github.com/fonttools/fonttools/issues/3825.
+        styleNameExists = nameTable.findMultilingualName(
```

Comment:
if findMultilingualName finds something that is either 2 or 17 and instance.coordinates == default, you can just set that to the instance.subfamilyNameID, no? Why call addMultilingualName again in that case? The latter will again try to find the same nameID you already found (the one in `styleNameExists`)